docs/metrics/bertscore.md

```diff
@@ -0,0 +1,117 @@
+    ```
+
+    We get the following BertScores: $P_\text{BERT} = 0.9526, R_\text{BERT} = 0.9480, F_\text{BERT} = 0.9503$. This is in-line with human judgement, as the reference and candidate texts are semantically very similar.
+
+    However, if we were to calculate the BLEU score given these candidates and references, BLEU would yield a sub-optimal score of $\text{BLEU} = 0.2403$, despite the sentences being the same, semantically. This shows an advantage of embeddings-based metrics over n-gram based metrics like BLEU.
+
+??? example "Semantically Different Texts"
+
+    This time, let our candidate and reference texts be:
+
+    | Candidate Text | Reference Text |
+    | --- | --- |
+    | The sun was setting behind the mountains. | This is a bad example |
+    | She walked along the beach, feeling the sand between her toes. | This has nothing to do with the other |
+    | The chef prepared a delicious meal with fresh ingredients. | Hello, world! |
+    | The old oak tree stood tall in the middle of the field. | Vivaldi is a classical conductor |
+    | The detective examined the clues carefully. | Wrong answer |
+
+    This yields BERTScores of: $P_\text{BERT} = 0.8328, R_\text{BERT} = 0.8428, F_\text{BERT} = 0.8377$.
+    Between different tasks, the baseline for a "good" BERTScore may vary based on different factors, like text length.
+
+## Advantages and Limitations
+
+BERTScore, originally designed to be a replacement to the BLEU score, is a powerful metric that closely aligns with human judgement. However, it comes with limitations.
+
+1. BERTScore is computationally expensive. The default model (```roberta-large```) used to calculate BERTScore requires 1.4GB of weights to be stored, and requires a forward pass through the model in order to calculate the score. This may be computationally expensive for large datasets, compared to BLEU which is straightforward and easy to compute. However, smaller distilled models like ```distilbert-base-uncased``` can be used instead to reduce the computational cost, at the cost of reduced alignment with human judgement.
+
+2. BERTScore is calculated using a black-box pretrained model. The score can not be easily explained, as the embedding space of BERT is a dense and complex representation that is only understood by the model. Though the metric provides a numerical score, it does not explain how or why the particular score was assigned. In contrast, n-gram-based metrics can easily be calculated by inspection.
+
+Limitations aside, BERTScore is still a powerful metric that can be included in NLP workflows to quantify the quality of machine-generated texts. It has been shown to have a high correlation with human judgement in tests, and is overall a better judge of similarity than BLEU. Furthermore, unlike BLEU, BERTScore's use of embeddings allows it to factor in context, semantics, and order into its score - which allows it to avoid the pitfalls of BLEU.
```

Comment:
It is weird to talk about `BLEU` alone. I'd recommend generalizing for `old fashioned text similarity metrics` or as you have written `n-gram-based metrics`